ReactViewResources/Loader/Public/ViewFrame.tsx

```diff
@@ -1,43 +1,37 @@
 ﻿import * as React from "react";
 import { IViewFrameProps } from "ViewFrame";
 import { newView, ViewMetadata } from "../Internal/ViewMetadata";
-import { ViewMetadataContext } from "../Internal/ViewMetadataContext";
-import { ViewSharedContext } from "./ViewSharedContext";
+import { getEnsureDisposeInnerViewsFlag, ViewMetadataContext } from "../Internal/ViewMetadataContext";
+import { ViewSharedContext} from "./ViewSharedContext";
+import {ViewPortal} from "../Internal/ViewPortal";
+import InternalViewFrameLegacy from "./ViewFrameLegacy";
 
-interface IInternalViewFrameProps<T> extends IViewFrameProps<T> {
+export interface IInternalViewFrameProps<T> extends IViewFrameProps<T> {
     viewMetadata: ViewMetadata;
-    context: any;
+    context?: T;
 }
 
 /**
  * Placeholder were a child view is mounted.
  * */
-export class ViewFrame<T> extends React.Component<IViewFrameProps<T>, {}, ViewMetadata> {
-
-    constructor(props: IViewFrameProps<T>, context: any) {
-        super(props, context);
+export function ViewFrame<T>(props: IViewFrameProps<T>): JSX.Element {
+    const viewMetadata = React.useContext(ViewMetadataContext);
+    const viewContext = React.useContext(ViewSharedContext);
+    
+    if(!getEnsureDisposeInnerViewsFlag()) {
+        return <InternalViewFrameLegacy viewMetadata={viewMetadata} context={viewContext} {...props} />;
     }
 
-    public render(): JSX.Element {
-        return (
-            <ViewMetadataContext.Consumer>
-                {viewMetadata =>
-                    <ViewSharedContext.Consumer>
-                        {viewcontext => <InternalViewFrame viewMetadata={viewMetadata} context={viewcontext} {...this.props} />}
-                    </ViewSharedContext.Consumer>
-                }
-            </ViewMetadataContext.Consumer>
-        );
-    }
+    return <InternalViewFrame viewMetadata={viewMetadata} context={viewContext} {...props} />;
 }
 
 class InternalViewFrame<T> extends React.Component<IInternalViewFrameProps<T>, {}, ViewMetadata> {
-
     private static generation = 0;
 
-    private generation: number;
+    private readonly generation: number;
```

Comment:
it's weird this is readonly 😅  
it's supposed to get incremented on new renders, no?